### ISSUE TITLE
API Misuse

### ISSUE THREAD

**Issue by mythodeia** [Neirys/YCHActionSheet#3]
Hello again,
i analysed the sample for memory leaks and got 2 messages inside method setupSectionViews:
API Misuse (Apple): Dictionary value cannot be nil
for previousButton & previousSection variables.
Compiler warns about specific instance where the loop is not executed at all so the variables stay nil.
Haven't had a crash so far, so i suppose its a false positive kind of case

**Comment by Neirys** [comment]
Hi, thanks for contributing on this project. I hope you enjoy it and can I hope it fit your needed.

Actually those warning are not leaks but leaving them unchanged could result in a crash in two cases :
- 1°) You try to display a empty YCHActionSheet (0 sections)
- 2°) You try to display YCHActionSheet when one of his section is empty (no title and no buttons).

I thought it was not a bad ideas to crash the app on those cases but after reconsideration, I should not be responsible for any crash, specially when crash error is as unexplicit as it was : `[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[0]`

I have changed the behaviour in the next commit. An empty YCHActionSheet or YCHActionSheetSection will be display with no contents.

I'm letting this issue open for now as I want some feedbacks from you whether this solution is acceptable or not.

**Comment by mythodeia** [comment]
@Neirys 
First of all i would like to thank you for your time and support on my issues. I really appreciate your help.
I have tested your commit just now and it works just fine. 

In my personal opinion this is the correct path to take when no sections have been added. 
Showing no content is far better than crashing on purpose.

Closing this issue as i think you have resolved the problem fine! If you think there is something else to offer for this issue reopen it and we can talk!

Thanks a lot again
